### PR TITLE
Improve testing experience with aptos names

### PR DESCRIPTION
**Pull request by gregnazario**
Ensure that there's a check that the reverse lookup registry actually exists when deploying to a non-mainnet scenario

```
aptos move test --package-dir core --named-addresses aptos_names=0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c,aptos_names_ad
min=0x91945b4672607a327019e768dd6045d1254d1102d882df434ca734250bb3581d,aptos_names_funds=0x78ee3915e67ef5d19fa91d1e05e60ae08751efd12ce58e23fc1109de87ea7865 --coverage

INCLUDING DEPENDENCY AptosFramework
INCLUDING DEPENDENCY AptosStdlib
INCLUDING DEPENDENCY AptosToken
INCLUDING DEPENDENCY MoveStdlib
BUILDING aptos_names
Running Move unit tests
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::time_helper::test_time_conversion
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::token_helper::test_combine_sub_and_domain_str
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::utf8_utils::test_latin_digits
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domains::test_time_is_expired
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::utf8_utils::test_utf8_to_vec_u64_fast
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::token_helper::test_combine_sub_and_domain_str_dom_only
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::token_helper::test_get_fully_qualified_domain_name
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::config::test_admin_config_requires_admin
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::price_model::test_price_for_domain_v1
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::config::test_cant_set_foundation_address_without_coin
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::price_model::test_scale_price_for_years
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::admin_can_force_create_domain_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::config::test_configs_are_set
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::admin_can_force_create_subdomain_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::config::test_default_token_configs_are_set
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::admin_can_force_seize_domain_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::config::test_default_tokens_configs_are_set
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::config::test_foundation_config_requires_admin
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::admin_can_force_set_name_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::admin_can_force_seize_subdomain_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::admin_can_force_set_subdomain_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::admin_force_seize_domain_name_doesnt_clear_unrelated_primary_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::admin_cant_force_create_subdomain_more_than_domain_time_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::clear_name_happy_path_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::dont_allow_double_domain_registrations_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::dont_allow_double_subdomain_registrations_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::dont_allow_rando_to_clear_domain_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::dont_allow_rando_to_clear_subdomain_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::dont_allow_rando_to_set_domain_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::e2e_test_with_invalid_signature
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::dont_allow_rando_to_set_subdomain_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::e2e_test_with_valid_signature
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::happy_path_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::happy_path_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::names_are_registerable_after_expiry_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::names_are_registerable_after_expiry_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::owner_can_clear_domain_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::rando_cant_force_create_subdomain_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::owner_of_expired_name_is_not_owner
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::rando_cant_force_create_domain_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::rando_cant_force_seize_subdomain_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::rando_cant_force_seize_domain_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::subdomain_e2e_tests::rando_cant_force_set_subdomain_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::rando_cant_force_set_name_address_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::set_primary_name_after_transfer_clears_old_primary_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::set_target_address_after_transfer_clears_old_primary_name_e2e_test
[ PASS    ] 0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domain_e2e_tests::test_register_domain_abort_with_disabled_unrestricted_mint
Test result: OK. Total tests: 47; passed: 47; failed: 0
+-------------------------+
| Move Coverage Summary   |
+-------------------------+
Module 867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::utf8_utils
>>> % Module coverage: 95.86
Module 867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::config
>>> % Module coverage: 94.68
Module 867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::verify
>>> % Module coverage: 100.00
Module 867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::token_helper
>>> % Module coverage: 94.66
Module 867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::time_helper
>>> % Module coverage: 100.00
Module 867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::price_model
>>> % Module coverage: 94.44
Module 867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c::domains
>>> % Module coverage: 85.16
+-------------------------+
| % Move Coverage: 90.12  |
+-------------------------+
Please use `aptos move coverage -h` for more detailed source or bytecode test coverage of this package
{
  "Result": "Success"
}
```

### CODE REVIEW

**Review by 0xmaayan — APPROVED** [review]
some formatting issues

File: core/sources/domains.move [hunk 1]
```diff
@@ -193,12 +205,21 @@ module aptos_names::domains {
 
     /// A wrapper around `register_name` as an entry function.
     /// Option<String> is not currently serializable, so we have these convenience methods
-    public entry fun register_domain(sign: &signer, domain_name: String, num_years: u8) acquires NameRegistryV1, RegisterNameEventsV1, ReverseLookupRegistryV1, SetNameAddressEventsV1, SetReverseLookupEventsV1 {
+    public entry fun register_domain(
+        sign: &signer,
```
[the remark this answers is not in the document]
Comment (gregnazario) [reply core/sources/domains.move:209]: This is the autoformatter from the Intellij plugin

File: core/sources/domains.move [hunk 2]
```diff
@@ -158,20 +160,30 @@ module aptos_names::domains {
     }
 
     public entry fun init_reverse_lookup_registry_v1(account: &signer) {
-        assert!(signer::address_of(account) == @aptos_names, error::permission_denied(ENOT_AUTHORIZED));
+        let addr = signer::address_of(account);
```
Comment (movekevin) [inline core/sources/domains.move:163]: Why? addr is only used once

**Review by movekevin — APPROVED** [review]

File: core/sources/domains.move (hunk 2 above)
Comment (gregnazario) [reply core/sources/domains.move]: Fixed.  probably was from a weird iteration

**Pull request merged**